### DISCUSSION TITLE
minimized windows pipe menu

### DISCUSSION THREAD

**Discussion by davidphilipbarr** [labwc/labwc#1663] [Show and tell]
I've quickly throw together a minimized window pipe menu using wlrctl - it's impossibly basic and i have doubts over the window matching credibility, but it might be useful/interesting now pipemenus are a thing!

```sh
echo '<openbox_pipe_menu>'
wlrctl toplevel list state:minimized  |
while read line; 
do 
appid=$(echo $line | cut -d ':' -f1)
apptitle=$(echo $line | cut -d ':' -f2-| sed '1s/.//' )
echo "<item label="\""$appid - $apptitle"\"">"
echo "<action name="\""Execute"\""><execute>"
echo "wlrctl window focus app_id:$appid state:minimized `title:'$apptitle'`"
echo "</execute></action></item>"
done
echo '</openbox_pipe_menu>'
```

**Comment by davidphilipbarr** [comment]
there are some backticks in there around the title and apptitle section showing as code

**Reply by johanmalm** [reply]
Nice script. I took the liberty of adding the triple back-ticks to make it a code-block. Hope that's okay.

**Reply by davidphilipbarr** [reply]
thanks, i've updated it a little and added to github: https://github.com/davidphilipbarr/labwcpipemenus/blob/main/app-list.sh
I'm still very much in the dark with how wlrctl works...

**Reply by salahoued** [reply]
Here is a link to some Openbox's pipemenus [openbox](http://openbox.org/wiki/Openbox:Pipemenus)
there is a calendar pipemenu for your panel-menu if you'd like although I don't use, but I started for it to make a calendar widget for `sfwbar`.

**Comment by salahoued** [comment]
Is pipemenu not supported in `labwc 0.7.1`, because I can't seem to get them to work?
But in the discussion #1776, I got to build the git version `0.7.1-149-g8d4f295` where pipemenus are working fine. is the full support planned for 0.7.2?

**Reply by Consolatis** [reply]
Yes, what is the current git version will become the next 0.7.2 release and that includes pipe menus. https://github.com/labwc/labwc/blob/master/NEWS.md should show most of the changes compared to 0.7.1.
